test/src/test/java/hudson/model/utils/AbortExceptionPublisher.java

```diff
@@ -0,0 +1,35 @@
+package hudson.model.utils;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Publisher;
+import hudson.tasks.Recorder;
+
+import java.io.IOException;
+
+/**
+ * Publisher that throws AbortException
+ */
+public class AbortExceptionPublisher extends Recorder {
```

Comment:
It is interesting question because i was unable follow all `phase` and `canContinue` from `perform` method logic. `BuildStepMonitor` seems also plays in this party. 
I can wrote more test but i have no idea how all this statuses should work. AbortException was just added and seems there is no any existed tests for this statuses.